### PR TITLE
Included docs for `feature_dropout` method and classes

### LINKED ISSUE #9886: [doc] FeatureAlphaDropout not documented

### PR DESCRIPTION
**Pull request by mtsokol**
Addresses #9886 

Hi @vadimkantorov!

Here's PR for linking `feature_dropout` method in docs and including `FeatureDropout` and `FeatureAlphaDropout` there also. 

Listing `feature_dropout` method required including it in `nn/functional.pyi.in` and providing implementation. Also for consistency I've added `FeatureDropout` class, as those methods have links to class implementations with more details.

Is it correct?

## Screenshots from built docs

1. `featuredropout` method:
featuredropout

2. Dropout classes list:
classes list

3. Additional `FeatureDropout` class:
featuredropout class

---

P.S. I think this should close #9886 as missing docs for `torch.default_generator` already have some PRs #39139 and #27847

### CODE REVIEW

**Comment by vadimkantorov** [comment]
@mtsokol For me it looks good, but I'm not on PyTorch team :)

I'm sure @mruberry or others would find the reviewer and help merging this in!

File: torch/nn/modules/dropout.py [hunk 1]
```diff
@@ -184,6 +184,46 @@ def forward(self, input: Tensor) -> Tensor:
...
 
+class FeatureDropout(_DropoutNd):
+    r"""Randomly masks out entire channels (a channel is a feature map,
+    e.g. the :math:`j`-th channel of the :math:`i`-th sample in the batch input
+    is a tensor :math:`\text{input}[i, j]`) of the input tensor).
+
+    Each channel will be zeroed out independently on every forward call with
+    probability :attr:`p` using samples from a Bernoulli distribution.
+
+    Internally used by `DropoutXd` making stack traces less confusing.
```
Comment (vadimkantorov) [inline torch/nn/modules/dropout.py:195]: I guess this may be omitted from the main docs or may be moved to some notes section

File: torch/nn/modules/dropout.py [hunk 2]
```diff
@@ -184,6 +184,46 @@ def forward(self, input: Tensor) -> Tensor:
...
+    will not regularize the activations and will otherwise just result
+    in an effective learning rate decrease.
+
+    Args:
+        p (float, optional): probability of an element to be zeroed. Default: 0.5
+        inplace (bool, optional): If set to ``True``, will do this operation
+            in-place
+
+    Shape:
+        - Input: :math:`(N, C, D, H, W)`
```
Comment (vadimkantorov) [inline torch/nn/modules/dropout.py:210]: What is semantics of D? Would FeatureDropout also work for 1-d case, i.e. NCT? 2-d case, i.e. NCHW?

File: torch/nn/modules/dropout.py [hunk 3]
```diff
@@ -184,6 +184,46 @@ def forward(self, input: Tensor) -> Tensor:
...
+        >>> m = nn.FeatureDropout(p=0.2)
+        >>> input = torch.randn(20, 16, 4, 32, 32)
+        >>> output = m(input)
+
+    .. _Efficient Object Localization Using Convolutional Networks:
+       https://arxiv.org/abs/1411.4280
+    """
+
+    def forward(self, input: Tensor) -> Tensor:
+        return F.feature_dropout(input, self.p, self.training)
```
Comment (vadimkantorov) [inline torch/nn/modules/dropout.py:224]: Does it support `inplace` Dropout constructor's argument?

**Comment by vadimkantorov** [comment]
Also, `Fixes #9886` is sentence in PR description problematic because it would case the bot to close the whole issue. If it can be closed (e.g. if torch.default_generator is also documented and there are no other standing items raised in that issue), better to ask in the issue

**Comment by mtsokol** [comment]
> Also, `Fixes #9886` wording in PR description is problematic because it would case the bot to close the whole issue. If it can be closed (e.g. if torch.default_generator is also documented and there are no other standing items raised in that issue), better to ask in the issue

Ah, sorry! It was present in PRs template so I thought I need to follow it. I changed it to "addresses".

File: torch/nn/modules/dropout.py (hunk 2 above)
Comment (mtsokol) [reply torch/nn/modules/dropout.py:210]: I followed `FeatureAlphaDropout` implementation which seems to differ only in [additionally applying additional factor to the noise](https://github.com/pytorch/pytorch/blob/df837d0384398f1fac1000189d62b05e7ff0240d/aten/src/ATen/native/Dropout.cpp#L56) and therefore I took shapes from there.

AFAIU `FeatureDropout` [just requires >=2 dimensionality](https://github.com/pytorch/pytorch/blob/df837d0384398f1fac1000189d62b05e7ff0240d/aten/src/ATen/native/Dropout.cpp#L14). Should I change it and also correct [existing FeatureAlphaDropout docs then](https://github.com/pytorch/pytorch/blob/df837d0384398f1fac1000189d62b05e7ff0240d/torch/nn/modules/dropout.py#L218)?

File: torch/nn/modules/dropout.py (hunk 3 above)
Comment (mtsokol) [reply torch/nn/modules/dropout.py:224]: I followed `FeatureAlphaDropout` implementation which seems to differ only in [additionally applying additional factor to the noise](https://github.com/pytorch/pytorch/blob/df837d0384398f1fac1000189d62b05e7ff0240d/aten/src/ATen/native/Dropout.cpp#L56) and on the first sight [it does support `inplace` parameter](https://colab.research.google.com/drive/1gS4dhmDkDdNhL_SRShpCv34lE_qaPajn?usp=sharing) but it looks that this parameter is not passed to function call [like here](https://github.com/pytorch/pytorch/blob/df837d0384398f1fac1000189d62b05e7ff0240d/torch/nn/modules/dropout.py#L142) and is never used.  

So changing this I could also change [AlphaDropout](https://github.com/pytorch/pytorch/blob/df837d0384398f1fac1000189d62b05e7ff0240d/torch/nn/modules/dropout.py#L184) and [FeatureAlphaDropout](https://github.com/pytorch/pytorch/blob/df837d0384398f1fac1000189d62b05e7ff0240d/torch/nn/modules/dropout.py#L233) if that's correct. But should `inplace` be dropped out of the docs or should it be passed to the function call?

File: torch/nn/modules/dropout.py (hunk 1 above)
Comment (mtsokol) [reply torch/nn/modules/dropout.py]: Sure! Removed
